doc/subsystems/networking/net-app-api.rst

```diff
@@ -0,0 +1,208 @@
+.. _net_app_api:
+
+Network Application API
+#######################
+
+The Network Application (net-app) API allows applications to:
+
+* **Initialize** the application for networking use. This means, for example,
+  that if the application needs to have an IPv4 address, and if DHCPv4 is
+  enabled, then the net-app API will make sure that the device will get an
+  IPv4 address before the application is started.
+
+* **Set** various options for the networking subsystem. This means that if the
+  user has set options like IP addresses, IEEE 802.15.4 channel etc. in the
+  project configuration file, then those settings are applied to the system
+  before the application starts.
+
+* **Create** a simple TCP/UDP server or client application. The net-app API
+  has functions that make it easy to create a simple TCP or UDP based network
+  application. The net-app API also provides transparent TLS and DTLS support
+  for the application.
+
+The net-app API functionality is enabled by :option:`CONFIG_NET_APP` option.
+The current net-app API implementation is still experimental and may change and
+improve in future releases.
+
+Initialization
+##############
+
+The net-app API provides a cpp:func:`net_app_init()` function that can
+configure the networking subsystem for the application. The following
+configuration options control this configuration:
+
+* :option:`CONFIG_NET_APP_AUTO_INIT` automatically configures the system
+  according to other configuration options. The user does not need to call
+  cpp:func:`net_app_init()` in this case as that function will be automatically
+  called when the system boots. This option is enabled by default.
+
+* :option:`CONFIG_NET_APP_INIT_TIMEOUT` specifies how long to wait for the
+  network configuration during the system boot. For example, if DHCPv4 is
+  enabled, and if the IPv4 address discovery takes too long or the DHCPv4
+  server is not found, the system will resume booting after this number of
+  seconds.
+
+* :option:`CONFIG_NET_APP_NEED_IPV6` specifies that the application needs IPv6
+  connectivity. The cpp:func:`net_app_init()` function will wait until it is
```

Comment:
whoops. all these :cpp:func: references are missing the leading colon, so aren't being processed correctly:

     :cpp:func:`net_app_init()`